### PR TITLE
AMQP 1.0 message format

### PR DESCRIPTION
**Pull request by wolf4ood**
This PR removes ntex-amqp-codec  and implements internal AMQP 1.0 codec

### CODE REVIEW

File: protocol/src/message/amqp/section.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        MessageAnnotations, Value,
+    },
+    AmqpDecodeError,
+};
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(clippy::large_enum_variant)]
+pub enum MessageSection {
+    Header(Header),
+    DeliveryAnnotations(DeliveryAnnotations),
```
Comment (Gsantomaggio) [inline protocol/src/message/amqp/section.rs:26]: The AMQP 1.0 fields we support are:
```
 public Annotations Annotations { get; internal set; }

        public ApplicationProperties ApplicationProperties { get; set; }

        public Properties Properties { get; set; }

        public Data Data { get; }
````
other fields _must_ be read-only, so only for decoding, see the [.NET client](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/Message.cs#L23-L29)

Other fields could cause incompatibility problems, see https://github.com/qweeze/rstream/issues/15

**Review by Gsantomaggio — CHANGES_REQUESTED** [review]

File: protocol/src/message/amqp/section.rs (hunk 1 above)
Comment (wolf4ood) [reply protocol/src/message/amqp/section.rs:26]: In this case the section is for decoding. Exposed fields for write are in 

`MessageBuilder` to recap we should support only

- data
- message_annotations
- properties
- application properties

when creating a message. 

The rest : 

- header
- footer
- delivery annotations
- amqp value
- amq sequence

just for decoding  right?

Comment (Gsantomaggio) [reply protocol/src/message/amqp/section.rs:26]: yes! that's correct!  
Thank you!

Comment (Gsantomaggio) [reply protocol/src/message/amqp/section.rs:26]: >in this case the section is for decoding. 


A ok! yes I was looking for the encoding part!

Comment (Gsantomaggio) [reply protocol/src/message/amqp/section.rs:26]: solved via https://github.com/rabbitmq/rabbitmq-stream-rust-client/pull/120/commits/4538d5f245c7781f9304675ad88e3ba286c095e5

**Comment by Gsantomaggio** [comment]
Thank you @wolf4ood will check asap!

**Pull request merged**